### PR TITLE
tests: update test to make snapd snap fixed twice

### PR DESCRIPTION
**Pull request by sergiocazzolato**
I see a bahaviour that snapd.failure fails when it needs to fix snapd
twice.
The first time it fixes correctly the snapd snap but the second one it
fails not giving much information about what failed.

### CODE REVIEW

File: tests/core/snapd-failover/task.yaml [hunk 1]
```diff
@@ -42,23 +42,27 @@ execute: |
...
-    unsquashfs -d ./snapd-broken "$SNAPD_SNAP"
-    echo "" > ./snapd-broken/usr/lib/snapd/snapd
-    (cd ./snapd-broken && snap pack .)
-    echo "Now install the broken snapd"
-    if snap install --dangerous ./snapd-broken/snapd_*.snap; then
-        echo "installing a broken snapd should not work, test broken"
-        exit 1
-    fi
+    for _ in $(seq 2); do
+        echo "Break snapd"
```
Comment (bboozzoo) [inline tests/core/snapd-failover/task.yaml]: The part that breaks the snapd snap can be done outside of the loop.

File: tests/core/snapd-failover/task.yaml [hunk 2]
```diff
@@ -42,23 +42,27 @@ execute: |
...
+        unsquashfs -d ./snapd-broken "$SNAPD_SNAP"
+        echo "" > ./snapd-broken/usr/lib/snapd/snapd
+        (cd ./snapd-broken && snap pack .)
+        echo "Now install the broken snapd"
```
Comment (bboozzoo) [inline tests/core/snapd-failover/task.yaml]: From here on, the lines should be part of the loop.

File: tests/core/snapd-failover/task.yaml [hunk 3]
```diff
@@ -42,23 +42,27 @@ execute: |
...
+        snap change --last=install|MATCH "there was a snapd rollback across the restart"
 
-    echo "restart snapd and ensure we can still talk to it"
-    systemctl restart snapd.socket snapd.service
-    snap list | MATCH "^snapd .* $current .*"
+        echo "restart snapd and ensure we can still talk to it"
+        systemctl restart snapd.socket snapd.service
+        snap list | MATCH "^snapd .* $current .*"
+
+        rm -rf ./snapd-broken
```
Comment (bboozzoo) [inline tests/core/snapd-failover/task.yaml]: This cleanup can be done outside of the loop, preferrably right after the snap is packed.

File: tests/core/snapd-failover/task.yaml [hunk 4]
```diff
@@ -42,23 +42,27 @@ execute: |
...
-    echo "Break snapd"
-    unsquashfs -d ./snapd-broken "$SNAPD_SNAP"
-    echo "" > ./snapd-broken/usr/lib/snapd/snapd
-    (cd ./snapd-broken && snap pack .)
-    echo "Now install the broken snapd"
-    if snap install --dangerous ./snapd-broken/snapd_*.snap; then
-        echo "installing a broken snapd should not work, test broken"
-        exit 1
-    fi
+    for _ in $(seq 2); do
```
Comment (bboozzoo) [inline tests/core/snapd-failover/task.yaml]: This could use a comment:
```
# make sure that failover can execute correctly twice in a row
```

**Comment by anonymouse64** [comment]
This change is included in my PR https://github.com/snapcore/snapd/pull/8169, but I need to do a little bit more in that PR before it can be merged I think

**Comment by sergiocazzolato** [comment]
This change is included in the PR #8169, so I am closing it

**Pull request closed without merging**